### PR TITLE
docs: man page modifications after removal of deprecated commands

### PR DESCRIPTION
**Pull request by DuckBoss**
I'm leaving this PR as a draft until the following PRs are merged:
 - **Remove 'redeem' command:** https://github.com/candlepin/subscription-manager/pull/3446
 - **Remove 'remove' command:** https://github.com/candlepin/subscription-manager/pull/3444
 - **Remove 'attach' command:** https://github.com/candlepin/subscription-manager/pull/3409
 - **Remove 'auto-attach' command:** https://github.com/candlepin/subscription-manager/pull/3451

This PR contains some redundant commits to visualize the final man page modifications after the above PRs are merged. I will rebase and squash this PR on top of the other man page changes, so there might not be much changes in this PR when all is said and done except some small modifications to language/wording.

Card ID: CCT-545

### CODE REVIEW

File: man/subscription-manager.8 [hunk 1]
```diff
@@ -297,87 +274,6 @@ command does two important things. Firstly, it will implicitly remove all of the
...
-
-
-.TP
-.B --servicelevel=LEVEL
-Sets the preferred service level to use with subscriptions automatically attached to the system. Service levels are commonly premium, standard, and none, though other levels may be available depending on the product and the contract. This option cannot be used when attaching specific pools via
-.B --pool
-or
-.B --file.
-
-.SS AUTO-ATTACH OPTIONS
```
Comment (ptoscano) [inline man/subscription-manager.8:337]: ah, note that auto-attach is (still?) not on the removal list -- we'll need to discuss about it

File: man/subscription-manager.8 [hunk 2]
```diff
@@ -77,60 +73,52 @@ must be passed as system arguments in a non-interactive session.
 2. unregister
 
 .IP
-4. auto-attach
```
Comment (ptoscano) [inline man/subscription-manager.8:80]: `auto-attach` is not removed (yet?)

Comment (jirihnidek) [reply man/subscription-manager.8:80]: No, it has not been removed yet.

**Review by ptoscano — APPROVED** [review]
This update LGTM, thanks.

Sadly the man page is in dire need for a much thorough review... :cry:

**Pull request merged**